src/legacy/ui/public/agg_types/controls/date_ranges.tsx

```diff
@@ -0,0 +1,184 @@
+
+import React, { Fragment, useState, useEffect } from 'react';
+import {
+  htmlIdGenerator,
+  EuiButtonIcon,
+  EuiButtonEmpty,
+  EuiFieldText,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiIcon,
+  EuiLink,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+import dateMath from '@elastic/datemath';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+import { isEqual, omit } from 'lodash';
+import { AggParamEditorProps } from 'ui/vis/editors/default';
+import { getDocLink } from '../../documentation_links';
+
+const generateId = htmlIdGenerator();
+const isEmpty = (value: any) => value === undefined || value === null || value === '';
+const validateDateMath = (value: string = '') => {
+  if (isEmpty(value)) {
+    return true;
+  }
+
+  const moment = dateMath.parse(value);
+  return moment != null && moment.isValid();
+};
+
+interface DateRangeValues {
+  from?: string;
+  to?: string;
+}
+
+interface DateRangeValuesModel extends DateRangeValues {
+  id: string;
+}
+
+function DateRangesParamEditor({
+  value = [],
+  setValue,
+  setValidity,
+}: AggParamEditorProps<DateRangeValues[]>) {
+  const [ranges, setRanges] = useState(() => value.map(range => ({ ...range, id: generateId() })));
+
+  // set up an initial range when there is no default range
+  useEffect(() => {
+    if (!value.length) {
+      onAddRange();
+    }
+  }, []);
+
+  useEffect(
+    () => {
+      // responsible for discarding changes
+      if (
+        value.length !== ranges.length ||
+        value.some((range, index) => !isEqual(range, omit(ranges[index], 'id')))
+      ) {
+        setRanges(value.map(range => ({ ...range, id: generateId() })));
+      }
+
+      setValidity(value.every(({ from, to }) => validateDateMath(from) && validateDateMath(to)));
+    },
+    [value]
+  );
+
+  const updateRanges = (rangeValues: DateRangeValuesModel[]) => {
+    // do not set internal id parameter into saved object
+    setValue(rangeValues.map(range => omit(range, 'id')));
+    setRanges(rangeValues);
+  };
+  const onAddRange = () => updateRanges([...ranges, { id: generateId() }]);
+  const onRemoveRange = (id: string) => updateRanges(ranges.filter(range => range.id !== id));
+  const onChangeRange = (id: string, key: string, newValue: string) =>
+    updateRanges(
+      ranges.map(range =>
+        range.id === id
+          ? {
+              ...range,
+              [key]: newValue === '' ? undefined : newValue,
+            }
+          : range
+      )
+    );
+
+  return (
+    <>
+      {ranges.map(({ from, to, id }) => (
+        <Fragment key={id}>
+          <EuiFlexGroup gutterSize="s" alignItems="center">
+            <EuiFlexItem>
+              <EuiFieldText
+                aria-label={i18n.translate('common.ui.aggTypes.dateRanges.fromColumnLabel', {
+                  defaultMessage: 'From',
+                  description:
+                    'Describes the beginning of a date range, e.g. *From* 2018-02-26 To 2018-02-28',
```

Comment:
Does this need the "Describes the" part?

Reply:
AFAIK, the `description` field was added by @timroes for translation vendors as a comment to `defaultMessage`. Not sure, do we still need it or do we need to customize it since it has not any functionality load

Reply:
I mean the actual words "Describes the...." I think the description can just start with "Beginning of a date range..."